perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/ResourcesManagerBlImpl.java

```diff
@@ -426,13 +430,21 @@ public void assignGroupsToResource(PerunSession perunSession, Iterable<Group> gr
 			}
 		}
 
+		// activate each group-resource assignment
+		for (Group g : groups) {
+			try {
+				activateGroupResourceAssignment(perunSession, g, resource, async);
```

Comment:
I think it is a good idea to reuse `activateGroupResourceAssignment` method, but now the same activation process (setting attributes,..) would be actually performed twice - once in `assignGroupsToResource` before this for cycle and once in `activateGroupResourceAssignment`. So consider using just `activateGroupResourceAssignment`. Also, be aware that this method silently skips activating assignments in `ACTIVE` state.